WcaOnRails/lib/results_validators/persons_validator.rb

```diff
@@ -111,6 +114,25 @@ def validate(competition_ids: [], model: Result, results: nil)
           if without_wca_id.select { |p2| p2.name == p.name }.length > 1 && !duplicate_newcomer_names.include?(p.name)
             duplicate_newcomer_names << p.name
           end
+          # Look for obvious person name issues
+          splt_name = p.name.split
+          if splt_name.any? { |n| n.downcase == n }
+            @warnings << ValidationWarning.new(:persons, competition_id,
+                                               LOWERCASE_NAME_WARNING,
+                                               name: p.name)
+          end
+          if splt_name.length > 2
+            if splt_name[0, splt_name.length-1].any? { |n| n.length == 1 }
+              @warnings << ValidationWarning.new(:persons, competition_id,
+                                                 MISSING_ABBREVIATION_PERIOD_WARNING,
+                                                 name: p.name)
+            end
+          end
+          if [' ', '.'].include?(p.name[1]) || ((p.name[-2] == " ") && !['I', 'V'].include?(p.name[-1])) || ((p.name[-1] == ".") && (p.name[-3] == " "))
```

Comment:
This `if`-check might be clever and efficient, but it looks really confusing and incredibly complex to the naked eye.

What about splitting the different `||` parts into separate variables with descriptive names first, and then on a separate line combine these?

```ruby
non_word_after_first_letter = [' ', '.'].include?(p.name[1])
descriptive_foo = (p.name[-2] == " ") && !['I', 'V'].include?(p.name[-1])
descriptive_bar = (p.name[-1] == ".") && (p.name[-3] == " ")

if non_word_after_first_letter || descriptive_foo || descriptive_bar
  ... do stuff ...
end
```

Reply:
Also, unrelated to my previous comment: Probably worth including a comment on why `I` and `V` are specifically excluded. I have a hunch, but not every person looking at this code might know.

Reply:
I split the expressions into variables, it's definitely a lot neater this way.

I initially didn't exclude I and V, but it turned out I had to in order to pass unit tests (fake names sometimes had numerical suffixes, which created unexpected warnings).